File: Creational/FactoryMethod/PHP/factories/MacWindow.php
<?php

require_once(__DIR__.'/../interfaces/IButton.php');
require_once(__DIR__.'/../products/MacButton.php');
require_once('AbstractWindow.php');

/**
 * Concrette factory.
 */

class MacWindow extends AbstractWindow
{
	/**
	 * @override
	 */
	protected function createButton(): IButton
	{
		return new MacButton();
	}
}

File: Creational/FactoryMethod/Java/factories/MacWindow.java
package patterns.creational.factorymethod.factories;

import patterns.creational.factorymethod.interfaces.IButton;
import patterns.creational.factorymethod.products.MacButton;

/**
 * Concrete factory.
 */

public class MacWindow extends AbstractWindow
{
	/**
	 * @Override
	 */
	protected IButton createButton()
	{
		return new MacButton();
	}
}
